src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -984,9 +961,9 @@
     (let [subject (get-csr-subject csr)]
       (sling/throw+
        {:kind :disallowed-extension
-        :msg (format (str "CSR '%s' contains subject alternative names (%s), which are disallowed. "
-                          "Use `puppet cert --allow-dns-alt-names sign %s` to sign this request.")
-                     subject (str/join ", " dns-alt-names) subject)}))))
+        :msg (format "%s %s"
+                     (tru "CSR ''{0}'' contains subject alternative names ({1}), which are disallowed." subject (str/join ", " dns-alt-names))
+                     (tru "Use `puppet cert --allow-dns-alt-names sign {0}` to sign this request." subject))}))))
```

Comment:
So, unfortunately for almost all of the places in this file where we are throwing an exception with a message string in it, I fear that we are catching it in a block like this:

https://github.com/janelu2/puppetserver/blob/056c17224fdc93f88f59b2ff8266a391a5fdad4d/src/clj/puppetlabs/services/ca/certificate_authority_core.clj#L52-L57

where it then gets both logged *and* returned in the HTTP response.

I'm not sure that I care about getting those all audited and fixed as part of this PR, because I don't even have a great idea of what the solution is going to end up being.  My best idea at the moment is to put two entries in this exception, like a ':msg-user' and a ':msg-system', and then have the catch block use the appropriate one for the appropriate thing.

If we decide to punt that to another PR then we need to capture a ticket about it.